libs/net/nodenetworkconfigurationpolicy.py

```diff
@@ -0,0 +1,183 @@
+from __future__ import annotations
+
+from dataclasses import asdict, dataclass
+from typing import Any
+
+from kubernetes.dynamic import DynamicClient
+from ocp_resources.exceptions import NNCPConfigurationFailed
+from ocp_resources.resource import Resource, ResourceEditor
+from timeout_sampler import retry
+
+from utilities.constants import TIMEOUT_5SEC, TIMEOUT_90SEC
+
+
+@dataclass
+class STP:
+    enabled: bool
+
+
+@dataclass
+class BridgeOptions:
+    stp: STP
+
+
+@dataclass
+class Port:
+    name: str
+
+
+@dataclass
+class Bridge:
+    options: BridgeOptions | None = None
+    port: list[Port] | None = None
+    allow_extra_patch_ports: bool | None = None
+
+
+@dataclass
+class IP:
+    auto_dns: bool = True
+    dhcp: bool = False
+    enabled: bool = False
+
+
+@dataclass
+class IPv4(IP):
+    pass
+
+
+@dataclass
+class IPv6(IP):
+    autoconf: bool = False
+
+
+@dataclass
+class Interfaces:
```

Comment:
I think the name should be singular.

Reply:
Just to be clear, the suggestion of @EdDev as I understand is:
```suggestion
class Interface:
```